virtual-classroom/src/main/java/ro/utcn/sd/mid/assign1/virtualclassroom/controller/ConsoleController.java

```diff
@@ -0,0 +1,360 @@
+            String loginRegisterCommand = scanner.next().trim().toLowerCase();
+            switch (loginRegisterCommand) {
+                case "login":
+                    try {
+                        print("Username: ");
+                        String username = scanner.next().trim();
+                        print("Password: ");
+                        String password = scanner.next().trim();
+                        loggedInUser = sOUserService.loginSOUser(username, password);
+                        loggedIn = true;    // exit while
+                    } catch (InvalidNameOrPasswordException e) {
+                        print("Invalid name or password! ");
+                    }
+                    break;
+
+                case "register":
+                    try {
+                        print("Username: ");
+                        String username = scanner.next().trim();
+                        print("Password: ");
+                        String password = scanner.next().trim();
+                        loggedInUser = sOUserService.registerSOUser(username, password);
+                        loggedIn = true;
+                    } catch (NameAlreadyExistsException e) {
+                        print("Name already taken, please choose a different one!");
+                    }
+                    break;
+                default:
+                    print("Invalid command.");
+            }
+        }
+
+
+        println("\nWelcome! Logged in as \"" + loggedInUser.getSOUsername() + "\".");
+        println("Type \"help\" to see a list of available commands");
+        loggedIn = false;
+        while (!loggedIn) {
+            print("\nCommand: ");
+            String command = scanner.next().trim();
+            switch (command) {
+                case "help":
+                    handleHelp();
+                    break;
+                case "list all":
+                    handleListAll();
+                    break;
+                case "ask":
+                    handleAsk();
+                    break;
+                case "filter tag":
+                    handleFilterTagTitle(true);
+                    break;
+                case "filter title":
+                    handleFilterTagTitle(false);
+                    break;
+                case "exit":
+                    loggedIn = true;
+
+                    // ================== Feature 2 ==================
+
+                case "list one":
+                    handleListOne();
+                    break;
+
+                case "answer":
+                    handleAnswer();
+                    break;
+
+                case "edit answer":
+                    handleEditDelete(true); // edit the answer
+                    break;
+
+                case "delete answer":
+                    handleEditDelete(false); // delete the answer
+                    break;
+
+                // ================== Feature 3 ==================
```

Comment:
The fact that you have to split your code into sections usually indicates that you should split your class into multiple sub-classes or your method into smaller methods ;)

Reply:
I thought it would have been easier to read and find the methods for each feature, but I understand.

Reply:
It is for sure easier to read with sections - but it would be even easier to read if it was split into smaller units of code ;)